### ISSUE TITLE
Site Logo Block doesn't load after previously uploading

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#32171]
## Description

With Gutenberg Version 10.7.0-rc.1, the Site Logo block now just shows as infinitely loading without an option to replace the image in the block toolbar. This is a regression from 10.6.2. 

## Step-by-step reproduction instructions

1. Use Gutenberg 10.6.2 and set a site logo in the Site Logo Block
2. Switch to Gutenberg Version 10.7.0-rc.1 and view where you previously added the Site Logo Block to see error. 

## Expected behaviour

Should be able to see the previous site logo that was uploaded and have the option to replace it. 

## Actual behaviour

The Site Logo Block doesn't show an option to replace it and continually shows as loading. 

## Screenshots or screen recording (optional)

https://user-images.githubusercontent.com/26996883/119401606-ebad8300-bc98-11eb-8e2c-4f4a8b8f198d.mov

## WordPress information
- WordPress version: 5.7.2
- Gutenberg version: Version 10.7.0-rc.1
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? TT1 Blocks

## Device information
- Device: Desktop
- Operating system: MacOS
- Browser: Chrome

**Comment by Mamaduka** [comment]
I think #32092 should also fix this issue.

I see a spinner in `trunk`, but when I switch to branch from the PR, Site Logo Block works as expected.

**Comment by colorful-tones** [comment]
I just found this same issue. However, I would like to add that this Issue's title: `Site Logo Block doesn't load after previously uploading` is not entirely suitable. As I've noticed that it happens in TwentyTwentyOne, with Gutenberg 10.7.1, and when just inserting a new Site Logo block in any Page/Post. 

However, if we're referring to `doesn't load after previously uploading` as in the Site Logo should always show the previous attached logo, regardless of where/when a new Site Logo block is placed. Then the title is likely suitable. 

Gif of issue in TwentyTwentyOne and Gutenberg 10.7.1
![Gutenberg-10 7 1-Site-Logo-block-bug_TwentyTwentyOne](https://user-images.githubusercontent.com/405912/119737205-e8f38f00-be4c-11eb-9da4-ade8dfd285ae.gif)
